src/components/dashboard/WhoContent.native.js

```diff
@@ -30,12 +30,10 @@ const WhoContent = ({ styles, setName, setPhone, error, text, value, next, state
     })
   }
 
-  const selectContact = (name, phone) => {
-    setName(name)
-    setPhone(phone)
-    setTimeout(() => {
-      return next()
-    }, 100)
+  const selectContact = async (name, phone) => {
+    await setName(name)
+    await setPhone(phone)
+    return setTimeout(next)
```

Comment:
i dont think setX is a promise.
you need to pass a setContact({name,phone}) instead of setName,setPhone
you dont need to pass "next"
on parent component you use useEffect(next,[contact])
@nicastelo @MathiasMoralesDixtra